pkg/apis/playlist/v0alpha1/legacy_storage.go

```diff
@@ -96,12 +101,12 @@ func (s *legacyStorage) Get(ctx context.Context, name string, options *metav1.Ge
 		UID:   name,
 		OrgId: info.OrgID,
 	})
-	if err != nil {
+	if err != nil || dto == nil {
+		if errors.Is(err, playlist.ErrPlaylistNotFound) || err == nil {
+			err = k8serrors.NewNotFound(s.SingularQualifiedResource, name)
```

Comment:
^^^ kubectl requires this error explicitly.  Strangely this was the cause for seeing error messages about "not found" printed while executing kubectl delete commands in https://github.com/grafana/grafana/pull/75709